clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -4342,10 +4342,17 @@ QualType Sema::CheckTemplateIdType(TemplateName Name,
     InstantiatingTemplate Inst(*this, TemplateLoc, Template);
     if (Inst.isInvalid())
       return QualType();
+    if (!AliasTemplate->getDeclContext()->isFileContext()) {
+      ContextRAII SavedContext(*this, AliasTemplate->getDeclContext());
```

Comment:
nit: This can be simplified using `std::optional`:
```cpp
std::optional<ContextRAII> C;
if (!A.getDeclContext()->isFileContext())
  C.emplace(*this, A->getDeclContext());
```

Reply:
> nit: This can be simplified using `std::optional`:
> 
> ```c++
> std::optional<ContextRAII> C;
> if (!A.getDeclContext()->isFileContext())
>   C.emplace(*this, A->getDeclContext());
> ```

Neat trick, I like it!

Reply:
Looks more clear. Thanks for your guidance!